plugins/outputs/socket_writer/socket_writer.go

```diff
@@ -55,13 +58,39 @@ func (sw *SocketWriter) Connect() error {
 	}
 
 	var c net.Conn
-	if tlsCfg == nil {
-		c, err = net.Dial(spl[0], spl[1])
+
+	if spl[0] == "vsock" {
+		addrTuple := strings.SplitN(spl[1], ":", 2)
+
+		//check if address string has two tokens
+		if len(addrTuple) < 2 {
+			return fmt.Errorf("CID and/or port number missing")
+		}
+
+		//parse CID and port number from address string
+		// CID and port numner are 32 bit
+		// source: https://man7.org/linux/man-pages/man7/vsock.7.html
```

Comment:
See comment on input
```suggestion
		// Parse CID and port number from address string both being 32-bit
		// source: https://man7.org/linux/man-pages/man7/vsock.7.html
```